### DISCUSSION TITLE
Sharing Capability - Tenant Level - Separate OneDrive and SharePoint values

### DISCUSSION THREAD

**Discussion by Nibushi** [pnp/powershell#3027] [Q&A]
Is there a way to get/set OneDrive's SharingCapability independently from SharePoint's via PowerShell at the tenant level? For instance, if I have external sharing set to Anyone for SharePoint but OneDrive set to another value such as New and Existing guests, Existing guests or Only people in your organization, the SharingCapability value returned will be SharePoint's sharing level. I can't see a way to get/set the value for OneDrive independently at the tenant level. 

It doesn't look like this is possible but would be good to be confirmed either way.

I know that sharing can be configured on a per site/OneDrive basis but I am specifically looking to report on the External Sharing level for both SharePoint and OneDrive as set at the tenant level.

**Comment by AndersRask** [comment]
I think you have to use the `Microsoft.Online.SharePoint.PowerShell `module to get the value using Get-SPOSite and point it to the MySite Hosts: `Get-SPOSite -Identity https://yourtenant-my.sharepoint.com/ | select sharingcapability`
You can set it using PnP.PowerShell using `Set-PnPTenantSite -Identity https://yourtenant-my.sharepoint.com -SharingCapability $sharingCapability` but AFAICS `Get-PnPTenantSite -Identity https://yourtenant-my.sharepoint.com` does not pick up the correct value (it just say Disabled)

**Reply by Nibushi** [reply]
Thanks for your response. I'll take a look at that.

I assume (if it is possible) that being able to do this via PnP.PowerShell will need to happen as MSOL module is being deprecated. That's not a question for you to answer, just me thinking out loud :-)
